### PR TITLE
Fix deserialize order

### PR DESCRIPTION
**Pull request by vovythevov**
@thewtex, @jcfr PTAL

### CODE REVIEW

File: GenerateCLP/GenerateCLP.cxx [hunk 1]
```diff
@@ -972,8 +1012,41 @@ void GenerateTCLAPParse(std::ostream &sout, ModuleDescription &module)
...
+    {
+    if (groupsIt->GetLabel() == "Parameter Serialization")
+      {
+      groupsIt->AddParameter(parametersDeSerialize);
+      break;
+      }
+    }
+#endif
+
+  sout << "    /* Those two vectors are used to store the JSON deserialized value */" << EOL << std::endl;
```
Comment (thewtex) [inline GenerateCLP/GenerateCLP.cxx]: Those -> These

**Comment by thewtex** [comment]
Looks good.  Are positional arguments handled? Please add more tests to verify behavior. Please squash the "WIP" commit.

**Comment by vovythevov** [comment]
@thewtex: Thanks for your comments !
I uploaded a new patch with:
 - removed/squashed WIP commit
 - Improved testing (added testing and improve the test itself).

**Comment by thewtex** [comment]
Nice work on the implementation and in coming up with this approach. Thanks for including the description in the commit message :+1:

**Pull request merged**